src/eos-updater-poll-common.c

```diff
@@ -883,20 +921,28 @@ get_x86_hw_descriptors (GHashTable *hw_descriptors)
     }
 }
 
+static gchar *
+get_custom_descriptors_path (void)
+{
+  return eos_updater_dup_envvar_or ("EOS_UPDATER_TEST_UPDATER_CUSTOM_DESCRIPTORS_PATH",
+                                    NULL);
+}
+
 GHashTable *
 get_hw_descriptors (void)
 {
   GHashTable *hw_descriptors = g_hash_table_new_full (g_str_hash, g_str_equal,
                                                       g_free, g_free);
-
-  if (g_file_test (DT_COMPATIBLE, G_FILE_TEST_EXISTS))
-    { /* ARM */
-      get_arm_hw_descriptors (hw_descriptors);
-    }
+  g_autofree gchar *custom_descriptors = NULL;
+
+  custom_descriptors = get_custom_descriptors_path ();
+  if (custom_descriptors != NULL)
+    get_custom_hw_descriptors (hw_descriptors,
+                               custom_descriptors);
+  else if (g_file_test (DT_COMPATIBLE, G_FILE_TEST_EXISTS))
+    get_arm_hw_descriptors (hw_descriptors);
   else
-    { /* X86 */
-      get_x86_hw_descriptors (hw_descriptors);
-    }
+    get_x86_hw_descriptors (hw_descriptors);
```

Comment:
Hm, I thought that DT_COMPATIBLE's value is rather self-documenting. Or did you mean documenting that the device-tree stuff exists only on the ARM hardware?